ext/liquid_c/tokenizer.c

```diff
@@ -90,10 +99,13 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
                 if (c != '}')
                     continue;
                 token->type = TOKEN_TAG;
+                if(token->str[cursor - tokenizer->cursor - 3] == '-')
+                    token->rstrip = tokenizer->lstrip_flag = 1;
                 goto found;
             }
             // unterminated tag
-            cursor = tokenizer->cursor + 2;
+            cursor = tokenizer->cursor + 3;
```

Comment:
Why are you taking three characters for an unterminated tag.  E.g. this means `'{{ab'` would get tokenized into `['{{a', 'b']` instead of `['{{', 'ab']`.  If the intention was to include the whitespace control character in the unterminated token, then I don't see the point in doing so.